p2p/protocol/identify/obsaddr.go

```diff
@@ -418,6 +454,87 @@ func (oas *ObservedAddrManager) recordObservationUnlocked(conn network.Conn, obs
 	oas.addrs[localString] = append(oas.addrs[localString], oa)
 }
 
+// For a given transport Protocol (TCP/UDP):
+//
+// 1. If we have an activated address, we are behind an Cone NAT.
+// With regards to RFC 3489, this could be either a Full Cone NAT, a Restricted Cone NAT or a
+// Port Restricted Cone NAT. However, we do NOT differentiate between them here and simply classify all such NATs as a Cone NAT.
+//
+// 2. If four different peers observe a different address for us on outbound connections, we
+// are MOST probably behind a Symmetric NAT.
+//
+// Please see the documentation on the enumerations for `network.NATDeviceType` for more details about these NAT Device types
+// and how they relate to NAT traversal via Hole Punching.
+func (oas *ObservedAddrManager) emitAllNATTypes() {
+	var allObserved []*observedAddr
+	for k := range oas.addrs {
+		allObserved = append(allObserved, oas.addrs[k]...)
+	}
```

Comment:
Index again when you mean to iterate.... It's more idiomatic to write this as
```
for _, addrs := range oas.addrs {
  allObserved = append(allObserved, addrs...)
}
```
```

Reply:
More pertinent however, do we really need this copy?

Reply:
well, it's probably a micro-optimization to fix the copy.  Have fixed the looping .

Reply:
no you haven't, it's still there.